### PR TITLE
[Fixed #4468] - Handled empty app name state properly

### LINKED ISSUE #4468: [Bug]-[50]:
## Description

In app edit mode if user tries to change the app name, and in doing so removes the complete string- they get an error right away; even if they are not trying to save an empty string. Error should be shown only if user tries to save the empty app name. 

### Steps to reproduce the behavior:
1. Open any app in edit mode
2. Try to edit the app name
3. Remove the complete string
4. On top right you'll see an error toast

### Important Details
 - Version: Cloud

https://user-images.githubusercontent.com/13262095/118106331-1f67ed80-b3fb-11eb-80a3-587e3d84744a.mp4

### PR DESCRIPTION
**Pull request by pranavkanade**
## Description
 1. added warning for empty app name next to input text field
 2. On empty string save action, used last saved value
 
Fixes #4468 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Updated an existing cypress test to check if - it shows localized warning when user adds invalid app name
- Added a new cypress test case to test - when user tries to save invalid app name

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: bug/4468-empty-appname-warning 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.57 **(-0.01)** | 32.65 **(0)** | 29.74 **(0)** | 52.1 **(0)**
 :red_circle: | app/client/src/components/ads/EditableText.tsx | 68.81 **(-0.83)** | 41 **(-0.84)** | 81.82 **(0)** | 66.33 **(-0.34)**
 :green_circle: | app/client/src/pages/Editor/EditableAppName.tsx | 42.86 **(1.95)** | 33.33 **(0)** | 0 **(0)** | 50 **(2.63)**</details>

### CODE REVIEW

**Review by rishabhsaxena — DISMISSED** [review]
@pskanade 
I think the UI looks a bit odd with the error message. Should we omit the error message? Instead we could try with just the error styles for the input and an error toast

**Comment by pskanade** [comment]
@rishabhsaxena Removed the localized warning. Will show an error toast if user tries to save an empty string.

File: app/client/src/pages/Editor/EditableAppName.tsx [hunk 1]
```diff
@@ -50,18 +50,16 @@ export default function EditableTextWrapper(props: EditableTextWrapperProps) {
...
-        isEditingDefault={isEditingDefault}
-        isInvalid={(value: string) => {
-          if (value.trim() === "") {
-            Toaster.show({
-              text: "Application name can't be empty",
-              variant: Variant.danger,
-            });
-          }
-          return false;
+        invokeInvalidInputToast={() => {
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/EditableAppName.tsx:53]: NIT: should we rename this to: `onValidationError`?

**Review by rishabhsaxena — APPROVED** [review]

**Review by rishabhsaxena — DISMISSED** [review]

**Review by nayan-rafiq — APPROVED** [review]
LGTM

**Comment by pranavkanade** [comment]
PR has gone stale, will fix the issue with new PR

**Pull request closed without merging**